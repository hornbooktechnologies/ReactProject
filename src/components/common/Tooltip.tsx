import React from 'react'
import { useTranslation } from 'react-i18next'
import { TooltipModel } from '../../models/TooltipModel'
import { cssClassName } from '../../utils/CssConstants'

const Tooltip = ({prefix, previousDayValue, unit, customClass}:TooltipModel) => {
  const { i18n } = useTranslation();
  return (
    <>
      <div className={`${customClass ? customClass : "uk-transdorm-origin-center-bottom uk-tooltip-bottom-center uk-transform-origin-center-bottom"} uk-tooltip uk-animation-fade-up uk-animation-center uk-togglabe-enter uk-text-left font-10 uk-text-center font-normal`}>
        <span className={`${cssClassName.COLOR_GREY_BLUE_80} uk-display-block`}>
          {i18n.t('UNIT_OPERATION.PREVIOUS_DAY')}
        </span>
        <span className="font-10 uk-text-bold uk-display-block mt-5">{prefix}{previousDayValue}&nbsp;{unit}</span>
      </div>
    </>
  )
}

export default Tooltip
