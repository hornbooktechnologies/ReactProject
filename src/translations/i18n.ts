import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
