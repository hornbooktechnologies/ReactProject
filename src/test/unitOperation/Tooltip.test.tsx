import React from "react";
import renderer from "react-test-renderer";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";
import Tooltip from "../../components/common/Tooltip";

describe("Component: UnitOperation Tooltip", () => {
  it("UnitOperation Tooltip renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <Tooltip previousDayValue={12} unit='YEN/kWh'/>
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
