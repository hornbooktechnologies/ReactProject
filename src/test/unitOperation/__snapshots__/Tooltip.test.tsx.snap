// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: UnitOperation Tooltip UnitOperation Tooltip renders correctly 1`] = `
<div
  className="uk-transdorm-origin-center-bottom uk-tooltip-bottom-center uk-transform-origin-center-bottom uk-tooltip uk-animation-fade-up uk-animation-center uk-togglabe-enter uk-text-left font-10 uk-text-center font-normal"
>
  <span
    className="color-grey-blue-80 uk-display-block"
  >
    UNIT_OPERATION.PREVIOUS_DAY
  </span>
  <span
    className="font-10 uk-text-bold uk-display-block mt-5"
  >
    12
    Â 
    YEN/kWh
  </span>
</div>
`;
